hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/LruBlockCache.java

```diff
@@ -373,7 +373,7 @@ private Cacheable asReferencedHeapBlock(Cacheable buf) {
     if (buf instanceof HFileBlock) {
       HFileBlock blk = ((HFileBlock) buf);
       if (!blk.isOnHeap()) {
```

Comment:
Here also it will be if shared memory type of block

Reply:
At least the API name in HFileBlock we can keep like isSharedMemory() instead of this on heap vs off heap?  That will be more clear when reading parts of code like here.  WHy we do this clone.

Reply:
> At least the API name in HFileBlock we can keep like isSharedMemory() instead of this on heap vs off heap? 

Will impl this in the new patch. 

> WHy we do this clone.

Because it's a offheap one, while our LRU only keep heap block, so need a deep clone here, please see: HBASE-22127

Reply:
That was not a Q why we clone. :-)  My bad. I put a '.' in between.  I mean with that name change and using that API in other code places like above, it will be more clear for any new reader to understand why we do this clone.